tcms/rpc/tests/test_testexecution.py

```diff
@@ -382,6 +382,17 @@ def update_test_case_text(self):
         self.execution_1.case.text = "Text Updated"
         self.execution_1.case.save()
 
+    def test_update_with_status_changes_tested_by(self):
+        execution = TestExecutionFactory(
+            tested_by=self.user
+        )
+
+        self.rpc_client.TestExecution.update(execution.pk, {"status": self.status_positive.pk})
+        execution.refresh_from_db()
+
+        self.assertEqual(execution.tested_by, self.api_user)
+        self.assertEqual(execution.status, self.status_positive)
+
```

Comment:
Scenario looks good but you also need an assertion to make sure if status isn't specified then `tested_by` isn't changed. Either a new test method or modification of some of the existing ones. Good candidate seems to be lines 238-248 - drop the status parameter and add assertion that tested_by is still None after the update.

Reply:
done.